src/elements/Label/Label.js

```diff
@@ -49,27 +46,29 @@ function Label(props) {
     className
   )
 
-  const _props = {
-    className: classes,
-    onClick: handleClick,
-    ...rest,
-  }
-
-  const _detail = detail || detailLink
-  const detailComponent = (detailLink || onClickDetail) && 'a' || detail && 'div'
+  const DetailComponent = (detailLink || onDetailClick) && 'a' || 'div'
 
   const _children = createFragment({
     icon: iconPropRenderer(icon),
     image: imagePropRenderer(image),
     text,
     children,
-    detail: _detail && createElement(detailComponent, { className: 'detail', onClick: handleClickDetail }, _detail),
-    remove: (removable || onClickRemove) && <Icon className='delete' onClick={handleClickRemove} />,
+    detail: detail && (
+      <DetailComponent className='detail' onClick={handleDetailClick}>{detail}</DetailComponent>
+    ),
+    remove: (removable || onRemove) && (
+      <Icon className='delete' onClick={handleRemove} />
+    ),
   })
 
-  // Do not destructure createElement import
-  // react-docgen only recognizes a stateless component when React.createElement is returned
-  return React.createElement(_component, _props, _children)
+  const LabelComponent = image || link || onClick ? 'a' : 'div'
```

Comment:
Just stating that this is not immediate clear to me, or at least leaves me questioning if it works exactly as I expect. Take that as you will.

`(image || link || onClick) ? 'a' : 'div'`